From fea55e12ed0dbfca35e9235497ea2d2fc4c5cd36 Mon Sep 17 00:00:00 2001
From: Anatol Derksen <derksen@fs-net.de>
Date: Mon, 1 Jul 2024 18:30:02 +0200
Subject: tools:CACert: Improve certGen.sh

Improve call of tput in warn_certs_not_for_production function. Use -T
parameter to define TERM enviroment.

diff --git a/tools/CACertificates/certGen.sh b/tools/CACertificates/certGen.sh
index 084244e3b..9509509f3 100755
--- a/tools/CACertificates/certGen.sh
+++ b/tools/CACertificates/certGen.sh
@@ -45,13 +45,13 @@ function makeCNsubject()
 
 function warn_certs_not_for_production()
 {
-    [ ! -z "$TERM" ] && tput smso
-    [ ! -z "$TERM" ] && tput setaf 3
+    [ ! -z "$TERM" ] && tput -T$TERM smso
+    [ ! -z "$TERM" ] && tput -T$TERM setaf 3
     echo "Certs generated by this script are not for production (e.g. they have hard-coded passwords of '1234'."
     echo "This script is only to help you understand Azure IoT Hub CA Certificates."
     echo "Use your official, secure mechanisms for this cert generation."
     echo "Also note that these certs will expire in ${days_till_expire} days."
-    [ ! -z "$TERM" ] && tput sgr0
+    [ ! -z "$TERM" ] && tput -T$TERM sgr0
 }
 
 function generate_root_ca()
